semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -2158,6 +2158,9 @@ and m_pattern a b =
       m_ident a1 b1 >>= fun () -> m_id_info a2 b2
   | G.PatLiteral a1, B.PatLiteral b1 -> m_literal a1 b1
   | G.PatType a1, B.PatType b1 -> m_type_ a1 b1
+  | G.PatConstructor (a1, [ G.PatEllipsis _ ]), B.PatConstructor (b1, _)
+  | G.PatConstructor (a1, _), B.PatConstructor (b1, [ B.PatEllipsis _ ]) ->
+      m_name a1 b1
   | G.PatConstructor (a1, a2), B.PatConstructor (b1, b2) ->
       m_name a1 b1 >>= fun () -> (m_list m_pattern) a2 b2
```

Comment:
so really you want to replace (m_list m_pattern) with an m_constructor_arguments
that will use m_list_with_dots and m_pattern.